verde/utils.py

```diff
@@ -681,6 +683,37 @@ def kdtree(coordinates, use_pykdtree=True, **kwargs):
     return tree
 
 
+def fill_nans(grid):
+    """
+    This methos is responsible for fill the NaN values in the grid using the KNN algorithm.
+
+    Parameters
+    ----------
+    grid : :class:`xarray.DataArray`
+        A 2D grid with one or more data variables.
+    Returns
+    -------
+    grid : :class:`xarray.DataArray`
+        A 2D grid with the NaN values filled.
+    """
+
+    filled_grid = grid.copy()
+
+    not_nan_values = np.argwhere(~np.isnan(grid.values))
+    unknown_indices = np.argwhere(np.isnan(grid.values))
```

Comment:
Use the `verde.grid_to_table` function and then drop the NaNs from it. It's easier and preserves the coordinates of the grid.